extensions/data-plane/data-plane-framework/src/main/java/org/eclipse/dataspaceconnector/dataplane/framework/manager/TransferServiceSelectionStrategy.java

```diff
@@ -0,0 +1,32 @@
+package org.eclipse.dataspaceconnector.dataplane.framework.manager;
+
+import org.eclipse.dataspaceconnector.dataplane.spi.pipeline.TransferService;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataFlowRequest;
+
+import java.util.stream.Stream;
+
+/**
+ * Functional interface for selecting which of (potentially) multiple {@link TransferService}s to use
+ * for serving a particular {@link DataFlowRequest}.
+ */
+public interface TransferServiceSelectionStrategy {
+    /**
+     * Selects which of (potentially) multiple {@link TransferService}s to use
+     * for serving a particular {@link DataFlowRequest}.
+     *
+     * @param request          the request.
+     * @param transferServices any number of services which are able to handle the request. May be an empty {@link Stream}.
+     * @return the service to be used to serve the request, selected among the input {@code transferServices}, or {@code null} if the stream is empty or no service should be used.
+     */
+    TransferService chooseTransferService(DataFlowRequest request, Stream<TransferService> transferServices);
+
+    /**
+     * Default strategy: use first matching service. This allows integrators to select
+     * order preferred {@link TransferService}s in the classpath.
+     *
+     * @return the first service, or {@code null} if the stream is empty.
+     */
+    static TransferServiceSelectionStrategy selectFirst() {
```

Comment:
Maybe adding @Nullable annotation for the return here, as it might be null.

Or maybe better why not having a default implementation of this interface:

```
public classTransferServiceSelectFirstStrategy implements TransferServiceSelectionStrategy  {
   @Override
   TransferService chooseTransferService(DataFlowRequest request, Stream<TransferService> transferServices) {
      return transferServices.findFirst().orElse(null);
   }
}
```

Then in `DataPlaneFrameworkExtension`, the code would become:

```              
 .transferServiceSelectionStrategy(Objects.requireNonNullElseGet(transferServiceSelectionStrategy,
                       new TransferServiceSelectFirstStrategy()))
```

Reply:
Yes that was my first go, but in the end it felt a functional style was more compact and saved a file. Do you have a clear preference for creating an extra class?

Reply:
Added Nullable annotation to chooseTransferService, good tip

Reply:
> Yes that was my first go, but in the end it felt a functional style was more compact and saved a file. Do you have a clear preference for creating an extra class?

No it's just a personal preference :) 
I am fine if you feel this approach suits best